src/execution/riscv64/simulator-riscv64.h

```diff
@@ -125,11 +125,15 @@ inline int64_t mulhsu(int64_t a, uint64_t b) {
 }
 
 // Floating point helpers
+#define isBoxedF32(v) ((uint32_t)((v >> 32) + 1) == 0)
 #define F32_SIGN ((uint32_t)1 << 31)
 union u32_f32 {
   uint32_t u;
   float f;
 };
+inline int64_t boxF32(float v) { 
+  return ((uint64_t)-1 << 32) | bit_cast<int32_t>(v); 
```

Comment:
The operator precedence of cast and `<<` can be a bit confusing. `return (0xFFFF_FFFF_0000_000 | bit_cast<int32_t>(v))` would be more straightforward to understand